### PR TITLE
[Azure.Monitor.Ingestion] Separate Monitor Live Test Pipelines

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by nisha-bhatia** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/31259

File: sdk/monitor/Azure.Monitor.Ingestion/tests.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+trigger: none
+
+stages:
+  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
+    parameters:
+      ServiceDirectory: monitor
+      Artifacts:
+        - name: azure-monitor-ingestion
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Ingestion/tests.yml:8]: name and safename should use the .NET naming style used here - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/ci.yml#L44

File: sdk/monitor/Azure.Monitor.Ingestion/tests.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+          groupId: com.azure
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Ingestion/tests.yml:9]: This is Java specific and can be removed.

**Comment by JoshLove-msft** [comment]
I'm curious how you would run the pipelines manually now that we are adding library specific tests.yml files. I guess the question is how the pipeline names will be generated. 
@benbp do you happen to know?

File: sdk/monitor/tests.yml [hunk 3]
```diff
@@ -4,3 +4,10 @@ extends:
   template: ../../eng/pipelines/templates/stages/archetype-sdk-tests.yml
```
Comment (JoshLove-msft) [inline sdk/monitor/tests.yml:4]: Is this file still needed?

**Comment by benbp** [comment]
> I'm curious how you would run the pipelines manually now that we are adding library specific tests.yml files. I guess the question is how the pipeline names will be generated. @benbp do you happen to know?

I would suggest picking either the top level tests.yml (one pipeline), or the inner tests.yml (three pipelines) vs. having both, unless that's really desired.

The naming convention is based on the parent directory @JoshLove-msft, so for the following:

- `sdk/monitor/tests.yml` => `net - monitor - tests`
- `sdk/monitor/Azure.Monitor.Ingestion/tests.yml` => `net - Azure.Monitor.Ingestion - tests`

**Comment by JoshLove-msft** [comment]
> > I'm curious how you would run the pipelines manually now that we are adding library specific tests.yml files. I guess the question is how the pipeline names will be generated. @benbp do you happen to know?
> 
> I would suggest picking either the top level tests.yml (one pipeline), or the inner tests.yml (three pipelines) vs. having both, unless that's really desired.
> 
> The naming convention is based on the parent directory @JoshLove-msft, so for the following:
> 
> * `sdk/monitor/tests.yml` => `net - monitor - tests`
> * `sdk/monitor/Azure.Monitor.Ingestion/tests.yml` => `net - Azure.Monitor.Ingestion - tests`

Thank you!

File: sdk/monitor/tests.yml (hunk 3 above)
Comment (JoshLove-msft) [reply sdk/monitor/tests.yml:4]: Based on @benbp's comment, we can remove this one.

**Comment by JoshLove-msft** [comment]
I guess we also need to update the [live test dashboard ](https://dev.azure.com/azure-sdk/internal/_dashboards/dashboard/d2224fd5-beee-47a0-90df-b24b73fdc8e7) to point to the new pipelines. It doesn't look like we have permissions to edit it. @benbp can you help with this?

File: sdk/monitor/Azure.Monitor.Query/tests.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+trigger: none
+
+stages:
+  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Query/tests.yml:4]: This doesn't seem to be the right relative path.

Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Query/tests.yml:4]:
```suggestion
  - template: ../../../eng/pipelines/templates/stages/archetype-sdk-tests.yml
```

Comment (pallavit) [reply sdk/monitor/Azure.Monitor.Query/tests.yml:4]: There are two modes in the yml. The beginning as '/' means from the root folder while ../.. is the relative path. Both should work in this case.

**Comment by pallavit** [comment]
> I guess we also need to update the [live test dashboard ](https://dev.azure.com/azure-sdk/internal/_dashboards/dashboard/d2224fd5-beee-47a0-90df-b24b73fdc8e7) to point to the new pipelines. It doesn't look like we have permissions to edit it. @benbp can you help with this?

@hallipr may also know I believe he is the owner as well.

**Comment by pallavit** [comment]
We may also need to manually delete older pipelines

File: sdk/monitor/Azure.Monitor.Query/tests.yml (hunk 4 above)
Comment (JoshLove-msft) [reply sdk/monitor/Azure.Monitor.Query/tests.yml:4]: Hmm, I was basing it on this error - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1920999&view=results

Comment (pallavit) [reply sdk/monitor/Azure.Monitor.Query/tests.yml:4]: Ahh may be we do not need the '/' at all.

**Comment by JoshLove-msft** [comment]
> We may also need to manually delete older pipelines

I ran the prepare pipelines command but I guess that doesn't handle deleting unused pipelines (which probably makes sense). Do you know if we would need Eng Sys help to delete https://dev.azure.com/azure-sdk/internal/_apps/hub/ms.vss-build-web.ci-designer-hub?pipelineId=2545?

File: sdk/monitor/Azure.Monitor.Query/tests.yml (hunk 4 above)
Comment (JoshLove-msft) [reply sdk/monitor/Azure.Monitor.Query/tests.yml:4]: Actually, I think you were right about the path not being the issue (it should be fine as-is though in the .NET repo we seem to always use the relative path syntax). 

I think the issue is that in the .NET [archetype-sdk-tests.yml](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/pipelines/templates/stages/archetype-sdk-tests.yml), we don't have an artifacts parameter. I think we can try removing the `artifacts` from the parameters list here.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+trigger: none
+
+stages:
+  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
+    parameters:
+      ServiceDirectory: monitor
+      Artifacts:
+        - name: Azure.Monitor.OpenTelemetry.Exporter
+          safeName: AzureMonitorOpenetemetryExporter
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests.yml:9]:
```suggestion
```

File: sdk/monitor/Azure.Monitor.Query/tests.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    parameters:
+      ServiceDirectory: monitor
+      Artifacts:
+        - name: Azure.Monitor.Query
+          safeName: AzureMonitorQuery
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Query/tests.yml:9]:
```suggestion
```

File: sdk/monitor/Azure.Monitor.Ingestion/tests.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+trigger: none
+
+stages:
+  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
+    parameters:
+      ServiceDirectory: monitor
+      Artifacts:
+        - name: Azure.Monitor.Ingestion
+          safeName: AzureMonitorIngestion
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Ingestion/tests.yml:9]:
```suggestion
```

**Comment by pallavit** [comment]
> > We may also need to manually delete older pipelines
> 
> I ran the prepare pipelines command but I guess that doesn't handle deleting unused pipelines (which probably makes sense). Do you know if we would need Eng Sys help to delete https://dev.azure.com/azure-sdk/internal/_apps/hub/ms.vss-build-web.ci-designer-hub?pipelineId=2545?

Yes, @hallipr or @benbp should be able to help us there. We may want to archive it for now until everything is running smoothly for the week or so. I will let engsys give us guidance here on best approach forward.

**Comment by benbp** [comment]
> > > We may also need to manually delete older pipelines
> > 
> > 
> > I ran the prepare pipelines command but I guess that doesn't handle deleting unused pipelines (which probably makes sense). Do you know if we would need Eng Sys help to delete https://dev.azure.com/azure-sdk/internal/_apps/hub/ms.vss-build-web.ci-designer-hub?pipelineId=2545?
> 
> Yes, @hallipr or @benbp should be able to help us there. We may want to archive it for now until everything is running smoothly for the week or so. I will let engsys give us guidance here on best approach forward.

We can manually delete, which will soft delete for 30 days. Want me to do it now, or wait a week?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+trigger: none
+
+stages:
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests.yml:3]:
```suggestion
extends:
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests.yml [hunk 9]
```diff
@@ -0,0 +1,6 @@
...
+  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests.yml:4]:
```suggestion
    template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
```

**Review by JoshLove-msft — APPROVED** [review]

**Comment by JoshLove-msft** [comment]
Still not working, I think we need to pass the Project parameter in each of the test.yml files - https://github.com/Azure/azure-sdk-for-net/blob/main/eng/pipelines/templates/stages/archetype-sdk-tests.yml#L132

**Review by JoshLove-msft — APPROVED** [review]

**Comment by JoshLove-msft** [comment]
@benbp looks like check enforcer commands are not working. Is this a known issue with new version?

**Comment by JoshLove-msft** [comment]
> @JoshLove-msft the PR looks mergeable to me, so I think override worked?

Ah yes, was expecting to see the failed checks cleared, but you are right it is mergeable now.

**Comment by benbp** [comment]
> Ah yes, was expecting to see the failed checks cleared, but you are right it is mergeable now.

Yep, it doesn't erase anything, just changes the status of the "required check" (the devops checks are not listed as required checks since github doesn't support dynamic/variable checks).
